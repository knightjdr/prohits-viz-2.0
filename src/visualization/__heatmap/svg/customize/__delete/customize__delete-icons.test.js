import React from 'react';
import { shallow } from 'enzyme';

import deleteIcons from './customize__delete-icons';

const handleClick = jest.fn();
const mouseEnter = jest.fn();
const mouseLeave = jest.fn();

describe('Delete icons', () => {
  describe('for columns', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <div>
          { deleteIcons(5, 20, handleClick, 'col', mouseEnter, mouseLeave) }
        </div>,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should have five g elements', () => {
      expect(wrapper.find('g').length).toBe(5);
    });

    describe('mouse events', () => {
      let g;

      beforeAll(() => {
        g = wrapper.find('g').first();
      });

      it('should call handleClick', () => {
        handleClick.mockClear();
        g.simulate('click');
        expect(handleClick).toHaveBeenCalledWith(0, 'col');
      });

      it('should call mouseEnter', () => {
        mouseEnter.mockClear();
        g.simulate('mouseenter');
        expect(mouseEnter).toHaveBeenCalledWith(0, 'col');
      });

      it('should call mouseLeave', () => {
        mouseLeave.mockClear();
        g.simulate('mouseleave');
        expect(mouseLeave).toHaveBeenCalledWith(0, 'col');
      });
    });
  });

  describe('for rows', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <div>
          { deleteIcons(5, 20, handleClick, 'row', mouseEnter, mouseLeave) }
        </div>,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should have five g elements', () => {
      expect(wrapper.find('g').length).toBe(5);
    });

    describe('mouse events', () => {
      let g;

      beforeAll(() => {
        g = wrapper.find('g').first();
      });

      it('should call handleClick', () => {
        handleClick.mockClear();
        g.simulate('click');
        expect(handleClick).toHaveBeenCalledWith(0, 'row');
      });

      it('should call mouseEnter', () => {
        mouseEnter.mockClear();
        g.simulate('mouseenter');
        expect(mouseEnter).toHaveBeenCalledWith(0, 'row');
      });

      it('should call mouseLeave', () => {
        mouseLeave.mockClear();
        g.simulate('mouseleave');
        expect(mouseLeave).toHaveBeenCalledWith(0, 'row');
      });
    });
  });
});
