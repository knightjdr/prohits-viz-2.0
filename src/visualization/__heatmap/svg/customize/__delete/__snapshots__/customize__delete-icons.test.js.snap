// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete icons for columns should match snapshot 1`] = `
<div>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-0"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    transform="translate(2 0) scale(1.6)"
  >
    <circle
      cx="5"
      cy="5"
      fill="#ffffff"
      r="4"
    />
    <path
      d="M5,0C2.238,0,0,2.238,0,5c0,2.761,2.238,5,5,5c2.762,0,5-2.238,5-5C10,2.238,7.762,0,5,0z M7.452,6.312 c0.095,0.096,0.095,0.248,0,0.344l-0.8,0.795c-0.094,0.096-0.248,0.096-0.342,0L5,6.129L3.688,7.451 c-0.095,0.096-0.248,0.096-0.343,0L2.548,6.654c-0.095-0.096-0.095-0.248,0-0.344L3.871,5L2.548,3.688 c-0.095-0.095-0.095-0.248,0-0.343l0.798-0.798c0.095-0.095,0.248-0.095,0.343,0L5,3.872l1.312-1.323 c0.095-0.095,0.248-0.095,0.342,0l0.799,0.798c0.095,0.095,0.095,0.248,0,0.343L6.129,5L7.452,6.312z"
      fill="#f44336"
    />
  </g>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-1"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    transform="translate(22 0) scale(1.6)"
  >
    <circle
      cx="5"
      cy="5"
      fill="#ffffff"
      r="4"
    />
    <path
      d="M5,0C2.238,0,0,2.238,0,5c0,2.761,2.238,5,5,5c2.762,0,5-2.238,5-5C10,2.238,7.762,0,5,0z M7.452,6.312 c0.095,0.096,0.095,0.248,0,0.344l-0.8,0.795c-0.094,0.096-0.248,0.096-0.342,0L5,6.129L3.688,7.451 c-0.095,0.096-0.248,0.096-0.343,0L2.548,6.654c-0.095-0.096-0.095-0.248,0-0.344L3.871,5L2.548,3.688 c-0.095-0.095-0.095-0.248,0-0.343l0.798-0.798c0.095-0.095,0.248-0.095,0.343,0L5,3.872l1.312-1.323 c0.095-0.095,0.248-0.095,0.342,0l0.799,0.798c0.095,0.095,0.095,0.248,0,0.343L6.129,5L7.452,6.312z"
      fill="#f44336"
    />
  </g>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-2"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    transform="translate(42 0) scale(1.6)"
  >
    <circle
      cx="5"
      cy="5"
      fill="#ffffff"
      r="4"
    />
    <path
      d="M5,0C2.238,0,0,2.238,0,5c0,2.761,2.238,5,5,5c2.762,0,5-2.238,5-5C10,2.238,7.762,0,5,0z M7.452,6.312 c0.095,0.096,0.095,0.248,0,0.344l-0.8,0.795c-0.094,0.096-0.248,0.096-0.342,0L5,6.129L3.688,7.451 c-0.095,0.096-0.248,0.096-0.343,0L2.548,6.654c-0.095-0.096-0.095-0.248,0-0.344L3.871,5L2.548,3.688 c-0.095-0.095-0.095-0.248,0-0.343l0.798-0.798c0.095-0.095,0.248-0.095,0.343,0L5,3.872l1.312-1.323 c0.095-0.095,0.248-0.095,0.342,0l0.799,0.798c0.095,0.095,0.095,0.248,0,0.343L6.129,5L7.452,6.312z"
      fill="#f44336"
    />
  </g>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-3"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    transform="translate(62 0) scale(1.6)"
  >
    <circle
      cx="5"
      cy="5"
      fill="#ffffff"
      r="4"
    />
    <path
      d="M5,0C2.238,0,0,2.238,0,5c0,2.761,2.238,5,5,5c2.762,0,5-2.238,5-5C10,2.238,7.762,0,5,0z M7.452,6.312 c0.095,0.096,0.095,0.248,0,0.344l-0.8,0.795c-0.094,0.096-0.248,0.096-0.342,0L5,6.129L3.688,7.451 c-0.095,0.096-0.248,0.096-0.343,0L2.548,6.654c-0.095-0.096-0.095-0.248,0-0.344L3.871,5L2.548,3.688 c-0.095-0.095-0.095-0.248,0-0.343l0.798-0.798c0.095-0.095,0.248-0.095,0.343,0L5,3.872l1.312-1.323 c0.095-0.095,0.248-0.095,0.342,0l0.799,0.798c0.095,0.095,0.095,0.248,0,0.343L6.129,5L7.452,6.312z"
      fill="#f44336"
    />
  </g>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-4"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    transform="translate(82 0) scale(1.6)"
  >
    <circle
      cx="5"
      cy="5"
      fill="#ffffff"
      r="4"
    />
    <path
      d="M5,0C2.238,0,0,2.238,0,5c0,2.761,2.238,5,5,5c2.762,0,5-2.238,5-5C10,2.238,7.762,0,5,0z M7.452,6.312 c0.095,0.096,0.095,0.248,0,0.344l-0.8,0.795c-0.094,0.096-0.248,0.096-0.342,0L5,6.129L3.688,7.451 c-0.095,0.096-0.248,0.096-0.343,0L2.548,6.654c-0.095-0.096-0.095-0.248,0-0.344L3.871,5L2.548,3.688 c-0.095-0.095-0.095-0.248,0-0.343l0.798-0.798c0.095-0.095,0.248-0.095,0.343,0L5,3.872l1.312-1.323 c0.095-0.095,0.248-0.095,0.342,0l0.799,0.798c0.095,0.095,0.095,0.248,0,0.343L6.129,5L7.452,6.312z"
      fill="#f44336"
    />
  </g>
</div>
`;

exports[`Delete icons for rows should match snapshot 1`] = `
<div>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-0"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    transform="translate(0 2) scale(1.6)"
  >
    <circle
      cx="5"
      cy="5"
      fill="#ffffff"
      r="4"
    />
    <path
      d="M5,0C2.238,0,0,2.238,0,5c0,2.761,2.238,5,5,5c2.762,0,5-2.238,5-5C10,2.238,7.762,0,5,0z M7.452,6.312 c0.095,0.096,0.095,0.248,0,0.344l-0.8,0.795c-0.094,0.096-0.248,0.096-0.342,0L5,6.129L3.688,7.451 c-0.095,0.096-0.248,0.096-0.343,0L2.548,6.654c-0.095-0.096-0.095-0.248,0-0.344L3.871,5L2.548,3.688 c-0.095-0.095-0.095-0.248,0-0.343l0.798-0.798c0.095-0.095,0.248-0.095,0.343,0L5,3.872l1.312-1.323 c0.095-0.095,0.248-0.095,0.342,0l0.799,0.798c0.095,0.095,0.095,0.248,0,0.343L6.129,5L7.452,6.312z"
      fill="#f44336"
    />
  </g>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-1"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    transform="translate(0 22) scale(1.6)"
  >
    <circle
      cx="5"
      cy="5"
      fill="#ffffff"
      r="4"
    />
    <path
      d="M5,0C2.238,0,0,2.238,0,5c0,2.761,2.238,5,5,5c2.762,0,5-2.238,5-5C10,2.238,7.762,0,5,0z M7.452,6.312 c0.095,0.096,0.095,0.248,0,0.344l-0.8,0.795c-0.094,0.096-0.248,0.096-0.342,0L5,6.129L3.688,7.451 c-0.095,0.096-0.248,0.096-0.343,0L2.548,6.654c-0.095-0.096-0.095-0.248,0-0.344L3.871,5L2.548,3.688 c-0.095-0.095-0.095-0.248,0-0.343l0.798-0.798c0.095-0.095,0.248-0.095,0.343,0L5,3.872l1.312-1.323 c0.095-0.095,0.248-0.095,0.342,0l0.799,0.798c0.095,0.095,0.095,0.248,0,0.343L6.129,5L7.452,6.312z"
      fill="#f44336"
    />
  </g>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-2"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    transform="translate(0 42) scale(1.6)"
  >
    <circle
      cx="5"
      cy="5"
      fill="#ffffff"
      r="4"
    />
    <path
      d="M5,0C2.238,0,0,2.238,0,5c0,2.761,2.238,5,5,5c2.762,0,5-2.238,5-5C10,2.238,7.762,0,5,0z M7.452,6.312 c0.095,0.096,0.095,0.248,0,0.344l-0.8,0.795c-0.094,0.096-0.248,0.096-0.342,0L5,6.129L3.688,7.451 c-0.095,0.096-0.248,0.096-0.343,0L2.548,6.654c-0.095-0.096-0.095-0.248,0-0.344L3.871,5L2.548,3.688 c-0.095-0.095-0.095-0.248,0-0.343l0.798-0.798c0.095-0.095,0.248-0.095,0.343,0L5,3.872l1.312-1.323 c0.095-0.095,0.248-0.095,0.342,0l0.799,0.798c0.095,0.095,0.095,0.248,0,0.343L6.129,5L7.452,6.312z"
      fill="#f44336"
    />
  </g>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-3"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    transform="translate(0 62) scale(1.6)"
  >
    <circle
      cx="5"
      cy="5"
      fill="#ffffff"
      r="4"
    />
    <path
      d="M5,0C2.238,0,0,2.238,0,5c0,2.761,2.238,5,5,5c2.762,0,5-2.238,5-5C10,2.238,7.762,0,5,0z M7.452,6.312 c0.095,0.096,0.095,0.248,0,0.344l-0.8,0.795c-0.094,0.096-0.248,0.096-0.342,0L5,6.129L3.688,7.451 c-0.095,0.096-0.248,0.096-0.343,0L2.548,6.654c-0.095-0.096-0.095-0.248,0-0.344L3.871,5L2.548,3.688 c-0.095-0.095-0.095-0.248,0-0.343l0.798-0.798c0.095-0.095,0.248-0.095,0.343,0L5,3.872l1.312-1.323 c0.095-0.095,0.248-0.095,0.342,0l0.799,0.798c0.095,0.095,0.095,0.248,0,0.343L6.129,5L7.452,6.312z"
      fill="#f44336"
    />
  </g>
  <g
    className="delete-icon__circle"
    cursor="pointer"
    key="icon-4"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    transform="translate(0 82) scale(1.6)"
  >
    <circle
      cx="5"
      cy="5"
      fill="#ffffff"
      r="4"
    />
    <path
      d="M5,0C2.238,0,0,2.238,0,5c0,2.761,2.238,5,5,5c2.762,0,5-2.238,5-5C10,2.238,7.762,0,5,0z M7.452,6.312 c0.095,0.096,0.095,0.248,0,0.344l-0.8,0.795c-0.094,0.096-0.248,0.096-0.342,0L5,6.129L3.688,7.451 c-0.095,0.096-0.248,0.096-0.343,0L2.548,6.654c-0.095-0.096-0.095-0.248,0-0.344L3.871,5L2.548,3.688 c-0.095-0.095-0.095-0.248,0-0.343l0.798-0.798c0.095-0.095,0.248-0.095,0.343,0L5,3.872l1.312-1.323 c0.095-0.095,0.248-0.095,0.342,0l0.799,0.798c0.095,0.095,0.095,0.248,0,0.343L6.129,5L7.452,6.312z"
      fill="#f44336"
    />
  </g>
</div>
`;
